plugins/org.obeonetwork.m2doc/src/org/obeonetwork/m2doc/api/TemplateConfigurationServices.java

```diff
@@ -72,20 +82,46 @@ public ConfigurationServices getConfigurationServices() {
      * @return generation eObject with definitions template information.
      */
     public Generation addProperties(Generation generation, TemplateInfo templateInfo) {
+        Multimap<String, EPackage> packagesByName = ArrayListMultimap.create();
+        for (String uri : templateInfo.getPackagesURIs()) {
+            if (EPackage.Registry.INSTANCE.containsKey(uri)) {
+                EPackage p = EPackage.Registry.INSTANCE.getEPackage(uri);
+                if (p != null) {
```

Comment:
you can have situations where a registered package has no name (there are many bogus EMF registrations out there) you should make sure it will not make the whole process fail.